### PR TITLE
update developer guide

### PR DESCRIPTION
**Pull request by Acerizm**
1. add only 1 user story ( the other user stories to be split evenly with others in the future )
2. add 1 use case description that is linked to chosen user story `Filter persons by tag`
3. add a few non-functional requirements

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -304,20 +307,41 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 *{More to be added}*
 
+
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+3.  A student with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  A student can easily learn and use commands provided by MODCheck.
```
Comment (joellow88) [inline docs/DeveloperGuide.md:339]: Might be considered a DG 'bug' cos in the grading criteria: 
>Not scoped clearly (i.e., hard to decide when it has been met)

Will approve PR but might consider removing in future iterations

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -304,20 +307,41 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+5.  New MODCheck features should have simple to use commands.
```
Comment (joellow88) [inline docs/DeveloperGuide.md:340]: Same as above

**Review by joellow88 — APPROVED** [review]

**Comment by joellow88** [comment]
PR approved but might consider removing or respecifying line 4 and 5 of the NFR due to requirement being rather vague.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Acerizm) [reply docs/DeveloperGuide.md]: Noted!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Acerizm) [reply docs/DeveloperGuide.md]: Noted!

**Comment by Acerizm** [comment]
@joellow88 Could you help me check if the 4th NFR is better now? Thanks!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -304,20 +307,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 *{More to be added}*
 
+
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+3.  A student with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  A student should be able to learn and use commands provided by MODCheck easily within an hour of usage.
```
Comment (joellow88) [inline docs/DeveloperGuide.md:339]: Sounds good!

**Review by joellow88 — APPROVED** [review]
Reworded NFR 4 sounds good!

**Comment by Acerizm** [comment]
@Gibson0918 @joellow88 Please help me do a quick check on the new merge where Gibson's work is merged with mine!

**Review by elvern18 — APPROVED** [review]

**Pull request merged**